autohooks/utils.py

```diff
@@ -131,5 +131,7 @@ def is_split_env():
         is_split = True
     except subprocess.CalledProcessError:
         is_split = False
+    except FileNotFoundError:
+        is_split = False
```

Comment:
I currently don't understand why subprocess should raise a FileNotFoundError error but maybe Windows is just different...

This change can be simplified by using
```suggestion
    except (subprocess.CalledProcessError, FileNotFoundError):
        is_split = False
```

Reply:
Thanks, I will make the change accordingly, in the meanwhile, posting trace here:
```
Traceback (most recent call last):
  File "C:\Users\ali\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ali\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\Scripts\autohooks.exe\__main__.py", line 7, in <module>
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\cli\__init__.py", line 108, in main
    args.func(term, args)
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\cli\activate.py", line 33, in install_hooks
    config = load_config_from_pyproject_toml(pyproject_toml)
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\config.py", line 187, in load_config_from_pyproject_toml
    return AutohooksConfig.from_toml(pyproject_toml)
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\config.py", line 163, in from_toml
    return AutohooksConfig.from_dict(config_dict)
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\config.py", line 145, in from_dict
    mode=_gather_mode(autohooks_dict.get_value("mode")),
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\config.py", line 94, in _gather_mode
    if is_virtual_env and not is_split_env():
  File "C:\Users\ali\AppData\Local\pypoetry\Cache\virtualenvs\tidy3d-webapi-qWdOphEq-py3.8\lib\site-packages\autohooks\utils.py", line 124, in is_split_env
    subprocess.run(
  File "C:\Users\ali\AppData\Local\Programs\Python\Python38\lib\subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Users\ali\AppData\Local\Programs\Python\Python38\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\ali\AppData\Local\Programs\Python\Python38\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

Reply:
Thanks for the traceback! Could you also split the changes into two commits to get nice change entries for the release?
The First should have the subject:
`Fix : Fix an issue where is_split_env method is not runable on windows platform`
The second: `Fix: Change pre-commit shebang line when using poetry` or similar

Reply:
According to this error: the pyproject.toml file has not been found, if I read this error correctly. This means, that we do not need to except this here, but you need to make sure, that you provide this file - otherwise autohooks won't work correcly anyways, or am I wrong @bjoernricks

Reply:
Yes, I will spite this pr.

> According to this error: the pyproject.toml file has not been found, if I read this error correctly. This means, that we do not need to except this here, but you need to make sure, that you provide this file - otherwise autohooks won't work correcly anyways, or am I wrong @bjoernricks

The not found file is "/usr/bin/env -S echo True", instead of the pyproject.toml file. The command line in subprocess couldn't run on windows platform.